test/unit/Cardano/Wallet/CoinSelection/FeeSpec.hs

```diff
@@ -0,0 +1,333 @@
+            , csOuts = [14,14]
+            , csChngs = [5,5]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,18,19]
+            , fChngs = [6,2,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 0
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,18,19]
+            , csChngs = [4,1]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,18,19]
+            , fChngs = [6,2,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 1
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,18,19]
+            , csChngs = [3]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,17,19]
+            , fChngs = [6,3,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 0
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,17,19]
+            , csChngs = [4,2]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,17,19]
+            , fChngs = [6,3,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 1
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,17,19]
+            , csChngs = [4,2]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20,20]
+            , fOuts = [14,17,19]
+            , fChngs = [6,3,1]
+            , fExtraUtxo = []
+            , fFee = 3
+            , fDust = 2
+            }) (Right $ FeeOutput
+            { csInps = [20,20,20]
+            , csOuts = [14,17,19]
+            , csChngs = [4]
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20]
+            , fOuts = [17]
+            , fChngs = [3]
+            , fExtraUtxo = []
+            , fFee = 4
+            , fDust = 0
+            }) (Left $ CannotCoverFee 1)
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20]
+            , fOuts = [16,18]
+            , fChngs = [4,2]
+            , fExtraUtxo = []
+            , fFee = 6
+            , fDust = 0
+            }) (Right $ FeeOutput
+            { csInps = [20,20]
+            , csOuts = [16,18]
+            , csChngs = []
+            })
+
+        feeUnitTest (FeeFixture
+            { fInps = [20,20]
+            , fOuts = [16,18]
+            , fChngs = [4,2]
+            , fExtraUtxo = []
+            , fFee = 6
+            , fDust = 2
+            }) (Left $ CannotCoverFee 2)
+
+        feeUnitTest (FeeFixture
+            { fInps = [10]
+            , fOuts = [7]
+            , fChngs = [3]
+            , fExtraUtxo = [1]
+            , fFee = 5
+            , fDust = 0
+            }) (Left $ CannotCoverFee 2)
```

Comment:
maybe should be `CannotCoverFee 1` as `fChngs` and `fExtraUtxo` gives 4 ... now because of MaybeT usage signalling about missing coins before random picking started

Reply:
will be corrected in coming version